### ISSUE TITLE
Disable inlay hints when bounds are not satisfied for syntactic sugar

### ISSUE THREAD

**Issue by fee1-dead** [rust-analyzer/rust-analyzer#10714]
Example:

```rust
struct A;
let a = A?;
```

![image](https://user-images.githubusercontent.com/43851243/140651905-6d4ed8f0-e2bd-4cc6-87a1-eb5fbac5d8bd.png)

I would personally think inlay hints should be disabled for this type of erroneous code, it gets complex when one confuses `.await?` with `?.await`:

![image](https://user-images.githubusercontent.com/43851243/140652320-87ee4c77-9f70-4468-80d4-fa15d06dfc8c.png)

**Comment by HKalbasi** [comment]
They shows that something is wrong, and it is good, I think?

**Comment by flodiebold** [comment]
No, we should really treat this as an error type, since the projection doesn't actually exist. I'm not sure how to best do this, though.
